activerecord/test/cases/calculations_test.rb

```diff
@@ -623,6 +623,13 @@ def test_count_selected_arel_attribute
     assert_equal 4, Account.distinct.select(Account.arel_table[:firm_id]).count
   end
 
+  def test_count_selected_arel_attributes
+    # Only MySQL supports COUNT with multiple columns, and only with DISTINCT.
+    skip unless current_adapter?(:Mysql2Adapter, :TrilogyAdapter)
+
+    assert_equal 5, Account.distinct.select(Account.arel_table[:id], Account.arel_table[:firm_id]).count
```

Comment:
I can’t get this to work with aliased columns, e.g. with
```rb
    assert_equal 5, Account.distinct.select(Account.arel_table[:id].as("foo"), Account.arel_table[:firm_id]).count
```
I get
```
  1) Error:
CalculationsTest#test_count_selected_aliased_arel_attributes:
NoMethodError: undefined method `relation' for an instance of Arel::Nodes::As
    lib/active_record/relation/calculations.rb:649:in `block in select_for_count'
    lib/active_record/relation/calculations.rb:637:in `map'
    lib/active_record/relation/calculations.rb:637:in `select_for_count'
    lib/active_record/relation/calculations.rb:426:in `perform_calculation'
    lib/active_record/relation/calculations.rb:245:in `calculate'
    lib/active_record/relation/calculations.rb:102:in `count'
    test/cases/calculations_test.rb:637:in `test_count_selected_aliased_arel_attributes'
```
(tested on sha 6be1ce4191c8f2ec61066b4bdebb7c2847c76fbe)

I’ll see if I can open a proper bug report tomorrow.

Reply:
Using `count distinct` with `as` in mysql is a syntax error, so I think this even is not worth fixing?

Reply:
Actually the test also fails without `distinct`:
```rb
assert_equal 5, Account.select(Account.arel_table[:id].as("foo"), Account.arel_table[:firm_id]).count
```

Reply:
Using `count` without `distinct` on multiple columns is not a valid SQL syntax.